gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.groovy

```diff
@@ -72,9 +72,12 @@ class JavaServiceDistributionPlugin implements Plugin<Project> {
             initScript.configure(distributionExtension.serviceName)
         }
 
-        CreateCheckScriptTask checkScript = project.tasks.create('createCheckScript', CreateCheckScriptTask)
-        project.afterEvaluate {
-            checkScript.configure(distributionExtension.serviceName, distributionExtension.checkArgs)
+        CreateCheckScriptTask checkScript = project.tasks.create('createCheckScript', CreateCheckScriptTask) {
+            group JavaServiceDistributionPlugin.GROUP_NAME
+            description "Generates healthcheck (service/monitoring/bin/check.sh) script."
+
+            serviceName { distributionExtension.serviceName }
```

Comment:
a slight alternative here would be to make `distributionExtension#serviceName` itself a `Supplier<String>` and then we can pass it around as a supplier instead of as a closure. Not totally clear what the right thing is here, but this gets rid of all the goofy afterEvaluate stuff.